2249/2/obscenity_tracker.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'terminal-table'
+require_relative 'obscenity_helper'
+
+# Tracker of obscene words
+class ObscenityTracker
+  DELIMITER_PATTERN = /#|против|vs|aka|VS/
+  HEADERS = [
+    'raper', 'battle', 'filthy', 'words in battle', 'words in round'
+  ].freeze
+
+  include ObscenityHelper
+
+  def initialize(directory, top_bad_words)
+    @directory = directory
+    @top_bad_words = top_bad_words
+    @artists_hash = {}
+    @current_file = nil
+  end
+
+  def run
+    puts Terminal::Table.new headings: HEADERS, rows: generate_terminal_rows
```

Comment:
Let's be more verbose.
I see `#generate_terminal_rows` uses `#generate_artists_hash`
So why not to place it as the first expression in your `#run` method?
Something like

```ruby
def run
  generate_artists_hash
  ...

  puts Terminal::Table.new heading #...
end
```